src/schemas/Polymorphic.js

```diff
@@ -45,7 +45,7 @@ export default class PolymorphicSchema {
     if (!this.isSingleSchema && !schemaKey) {
       return value;
     }
-    const id = isImmutable(value) ? value.get('id') : value.id;
+    const id = this.isSingleSchema ? void 0 : isImmutable(value) ? value.get('id') : value.id;
```

Comment:
To avoid confusion for others, since `void 0` isn't common, can we use a canonical `undefined` here?

Reply:
Sure, I've just updated the patch and rebased it to upstream/master.